File: 45/solution.go
package _45

func jump(nums []int) int {
	maxPos := 0
	end := 0
	step := 0
	n := len(nums)
	for i := 0; i < n-1; i++ {
		if maxPos >= i {
			maxPos = max(maxPos, i+nums[i])
			if i == end {
				end = maxPos
				step++
			}
		}
	}
	return step
}

func max(a, b int) int {
	if a > b {
		return a
	}
	return b
}

File: 45/Solution.java
class Solution {
    public int jump(int[] nums) {
        int maxPos = 0;
        int end = 0;
        int step = 0;
        for (int i = 0; i < nums.length - 1; i++) {
            if (maxPos >= i) {
                maxPos = Math.max(maxPos, i + nums[i]);
                if (i == end) {
                    end = maxPos;
                    step++;
                }
            }
        }
        return step;
    }
}